Mathlib/LinearAlgebra/Matrix/GeneralLinearGroup.lean

```diff
@@ -329,4 +331,98 @@ def planeConformalMatrix {R} [Field R] (a b : R) (hab : a ^ 2 + b ^ 2 ≠ 0) :
+    finrank_span_eq_card s.property, Fintype.card_fin]
+
+def inductiveStepEquiv {k : ℕ} :
+    { s : Fin (k + 1) → Fin n → 𝔽 // LinearIndependent 𝔽 s } ≃
+      Σ s : { s : Fin k → Fin n → 𝔽 // LinearIndependent 𝔽 s },
+        ((Submodule.span 𝔽 (Set.range (s : Fin k → Fin n → 𝔽)))ᶜ : Set (Fin n → 𝔽)) where
+  toFun s := by
+    have := linearIndependent_fin_succ.mp s.property
+    use ⟨Fin.tail s.val, this.left⟩
+    exact ⟨s.val 0, this.right⟩
+  invFun s := by
+    use Fin.cons s.2.val s.1.val
+    exact linearIndependent_fin_cons.mpr ⟨s.1.property, s.2.property⟩
+  left_inv _ := by simp only [Fin.cons_self_tail, Subtype.coe_eta]
+  right_inv := fun ⟨_, _⟩ => by simp only [Fin.cons_zero, Subtype.coe_eta, Sigma.mk.inj_iff,
+    Fin.tail_cons, heq_eq_eq, and_self]
+
+lemma card_LinearInependent_subtype {k : ℕ} (hk : k ≤ n) :
+    Fintype.card { s : Fin k → (Fin n → 𝔽) // LinearIndependent 𝔽 s } =
+      ∏ i : Fin k, ((q) ^ n - (q) ^ i.val) := by
+  induction' k with k ih
+  · simp only [Nat.zero_eq, LinearIndependent, Finsupp.total_fin_zero, LinearMap.ker_zero,
+    Fintype.card_ofSubsingleton, Finset.univ_eq_empty, Finset.prod_empty]
+  · simp only [Fintype.card_congr (inductiveStepEquiv n), Fintype.card_sigma, complement_card n, Finset.sum_const, Finset.card_univ, smul_eq_mul, ih (Nat.le_of_succ_le hk), mul_comm,
+    Fin.prod_univ_succAbove _ k, Fin.natCast_eq_last, Fin.val_last, Fin.succAbove_last,
+    Fin.coe_castSucc]
+
+lemma eq_matrix_basis (M : Matrix (Fin n) (Fin n) 𝔽) :
+    M = Basis.toMatrix (Pi.basisFun 𝔽 (Fin n)) (transpose M) := rfl
+
+/-- Equivalence between `GL n F` and `n` vectors of length `n` that are linearly independent. Given
+by sending a matrix to its coloumns. -/
+noncomputable def equiv_GL_linearindependent {F : Type*} [Field F] (hn : 0 < n) :
+    GL (Fin n) F ≃ { s : Fin n → (Fin n → F) // LinearIndependent F s } where
+  toFun M := ⟨transpose M, by
+    apply linearIndependent_iff_card_eq_finrank_span.2
+    rw [Set.finrank, ← rank_eq_finrank_span_cols, rank_unit]⟩
+  invFun M := by
+    apply GeneralLinearGroup.mk'' (transpose (M.1))
+    rw [eq_matrix_basis n (transpose (M.1)), transpose_transpose]
+    have : Nonempty (Fin n) := Fin.pos_iff_nonempty.1 hn
+    have hdim : Fintype.card (Fin n) = FiniteDimensional.finrank F (Fin n → F) := by
+      simp only [Fintype.card_fin, FiniteDimensional.finrank_fintype_fun_eq_card]
+    let b := basisOfLinearIndependentOfCardEqFinrank M.2 hdim
+    rw [show M = ⇑b by simp only [b, coe_basisOfLinearIndependentOfCardEqFinrank]]
+    have : Invertible ((Pi.basisFun F (Fin n)).toMatrix ⇑b) :=
+      (Pi.basisFun F (Fin n)).invertibleToMatrix b
+    exact isUnit_det_of_invertible _
+  left_inv := by
+    intro
+    ext
+    simp only [transpose_transpose]
+    exact rfl
+  right_inv := by exact congrFun rfl
+
+noncomputable instance fintype : Fintype (GL (Fin n) 𝔽) := by
+    exact Fintype.ofFinite (GL (Fin n) 𝔽)
+
```

Comment:
Again the bot is complaining that the indentation isnt correct. I've fixed it in the suggestion.